include/nuttx/timers/oneshot.h

```diff
@@ -266,7 +266,10 @@ int oneshot_max_delay(FAR struct oneshot_lowerhalf_s *lower,
   clock_t tick;
   int ret;
 
-  DEBUGASSERT(lower->ops->tick_max_delay);
+  if (lower->ops->tick_max_delay == NULL)
+    {
+      return -ENOSYS;
```

Comment:
In a traditional Unix system, ENOSYS is returned only if an entire system call is not supported.  ENOSUPP is the correct error return if a feature or function of a system call is not supported.

Reply:
Done